File: problems/problem 8/test_cases.c
#include <stdio.h>


long long fib(int n);

int failures=0;

void check(int n, long long expected) {
    long long output=fib(n);
    if (output==expected) {
        printf("PASS: n=%d\n", n);
    } else {
        printf("FAIL: n=%d Expected=%lld Got=%lld\n", n, expected, output);
        failures++;
    }
}

int main() {
    check(10,55);
    check(20,6765);
    check(45,1134903170LL);

    if (failures>0) return 1;
    return 0;
}

File: problems/problem 4/test_cases.java
public class test_cases {
    private static int failures = 0;

    private static void check(int[][] grid, int expected) {
        int output = solution.maxPathSum(grid);
        if (output==expected) {
            System.out.println("PASS");
        } else {
            System.out.println("FAIL Expected=" + expected + " Got=" + output);
            failures++;
        }
    }

    public static void main(String[] args) {
        check(new int[][]{{1,2,3,4},{2,2,1,1},{5,1,1,1}},11);
        check(new int[][]{{-1,2},{1,3}},4);
        check(new int[][]{{1,1,1,1},{1,1,1,1},{1,1,1,1},{1,1,1,9}},15);

        if (failures>0) System.exit(1);
        else System.exit(0);
    }
}
